### ISSUE TITLE
Improve motion controls on Quest like controllers when a tool is taking control of one of the elements

### ISSUE THREAD

**Issue by Stellanora64** [Yellow-Dog-Man/Resonite-Issues#2339]
### Is your feature request related to a problem? Please describe.

When a tooltip is equipped on a quest like controller the locomotion is locked to one controller and becomes tank controls. This can be very disorientating and cumbersome when developing with tool tips.

Another issue this causes is when two tooltips are equipped in both hands the user becomes stuck, as both joysticks movements are locked. 

### Describe the solution you'd like

As described in #2338 by Froox: 
> Decoupling from the axis is potentially feasible. It's a complicated problem.
> We generally recommend to focus on the problem, rather than specific solutions and let us work out the solution

So I will leave the specifics of the solution to this request to the developers.

### Describe alternatives you've considered

Leave as it currently is or the use of Mods.

### Additional Context

This issue is made as a less specific solution follow up to #2338 , which itself was made due to the following issues:
#125 #2310 #1817 #17

### Requesters

Stellanora

**Comment by Azavit** [comment]
This is a very big issue... almost every single person who I have introduced to the platform using the Quest/Quest like controllers asks if there is a way to fix this.

It doesn't effect me specifically but the people I've brought into this platform using quest/quest like controls always ask about this and more often than not states that it is "unplayable with out the mod" once using it.

**Comment by ko-tengu** [comment]
Yeah, it's a massive roadblock to new users and it's just extremely frustrating to work around as a Quest user even for the most casual editing or socializing. If the NoTankControls mod ever stopped working, I would honestly write the platform off as unplayable for myself because it is such a constant frustration for nearly every single layer of interaction.

**Comment by Frooxius** [comment]
I do have a few questions to understand some parts of this better:
1) You mention it affects socializing - why is that so? This shouldn't generally be an issue unless you're using building tools. 
2) How does the mod deal with the issue of overloaded inputs? E.g. so you don't move/jump when you select things with the dev tooltip for example

**Comment by ko-tengu** [comment]
It affects socializing because a huge amount of random gadgets and toys use tooltips for their basic functions because it's the quickest way to give a user a thing to hold and one or two inputs to make funny sounds. Maybe there's an argument that the water gun could be just a trigger and an interaction point on the model instead of using the visible "add second button" checkbox, but that's the way most things are built and the obvious way to build them.

For the NoTankControls mod, as far as I know it simply and entirely disables the axis lock, so secondary buttons are unchanged, but if a tool ever does show up that uses the joystick axis it may interfere. The only tool I've ever used that has used the joystick axis intentionally is the laser for remote grabbing and rotating, and it does still block movement for that one.

**Comment by Zyzyl** [comment]
Since Froox asked about it, here's the original No Tank Controls mod https://github.com/furrz/NoTankControls
This is fork with some additional functionality https://github.com/Nytra/NoTankControls/

**Comment by SlyTheFloof** [comment]
me personally i don't mind that much about the tank controls hoping there would be a toggle for it for thoes that don't need / want it

**Comment by Frooxius** [comment]
There already is - you can disable "Use Secondary" on custom tools. 

The issue is that with how the system works, some tools currently require it to be on, because they do use secondary.

**Comment by troyBORG** [comment]
#you can disable "Use Secondary" on custom tools. 

Is that in the settings menu?

**Comment by Frooxius** [comment]
> > There already is - you can disable "Use Secondary" on custom tools.
> 
> Is that in the settings menu?

No, it's on the RawDataTooltip. You can disable it for your own tools.

**Comment by troyBORG** [comment]
Oh I was hoping it could of been an global override.   
Like I was in a world Quake Arena world, and we don't have perms in there to do that.
I couldn't move correctly in there so I was at a disadvantage compared to people who was using the Mod.

I also cloudspawn my devtip so I can't really edit it to force that setting.

**Comment by Stellanora64** [comment]
That's already been discussed in #1817

**Comment by troyBORG** [comment]
> Also that setting doesn't exist for dev tips, so it doesn't really matter if it's cloud spawned or not.

How are they doing it with the mod then?  Whatever they are doing there is what we need...

**Comment by Stellanora64** [comment]
I'm not 100% certain on this as I haven't looked at their source code, but I believe they decouple seconds action, and secondary axis (currently the same thing as far as resonite is concerned). They then just disable secondary axis and the movement override for tools that don't need it. with some other stuff to handle edge cases like anchors

**Comment by troyBORG** [comment]
Probably?  But yeah that's mostly kind of what I wanting.   Maybe an options in the settings [like I did a mock-up](https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/2339#issuecomment-2427892421) to pretty much "enable" what the mod is doing.

Then if there is that rare case where you find a tool that needs the secondary button.  You can go into settings and "disable" it so that item works.

**Comment by Frooxius** [comment]
My problem with that is that it creates more confusing behavior and it's clunky to work with. If you have tool that uses secondary and you try to use that tool, you're suddenly moving at the same time when trying to use the tool with those controllers. Having to go to menu and toggle it on and off is not user friendly.

We need to redesign the approach to tools a bit there to solve this properly, to make sure we're not overloading the tools and perhaps to provide more fine grained control over what the tool actually uses.

**Comment by Redd56** [comment]
I agree its huge friction point. To the point that that whenever notankcontrolls (the mod) breaks o kinda take a break from the game.

Also most of the new users I've oriented recently (about 5 of em) run quest 2 or 3 or 3s meaning they have that issue. They all seem to be anoyed bu this issue

**Comment by Frooxius** [comment]
I see, that makes sense. We'll see about prioritizing this at some point for improving the initial user workflow, but one thing that's question is how does it compare against other frustration points for new users (because there's a bunch)?

**Comment by Nytra** [comment]
I would not want the thumbrest to be a secondary button because that's where I rest my thumb, so it could lead to accidental presses

**Comment by Redd56** [comment]
I guess that is true for some people. I wish it was used for something in game tho. Maybe jumping as clicking the joystick is anoying at times

**Comment by Nytra** [comment]
I actually have a custom binding the puts jump AND secondary on the right A button, it feels nicer

then dash menu is thumbstick click

**Comment by JackTheFoxOtter** [comment]
> I see, that makes sense. We'll see about prioritizing this at some point for improving the initial user workflow, but one thing that's question is how does it compare against other frustration points for new users (because there's a bunch)?

I think it's pretty high. Not as high as performance (because not being able to run the game > jank locomotion controls while building), but locomotion controls are pretty important. If controls are frustrating, that very quickly leads to users being frustrated with the entire experience I think. It's also the biggest frustration point I myself have in other VR games, when the controls don't feel good and intuitive, often the entire game doesn't.
